src/ipc/ipc4/handler.c

```diff
@@ -1330,9 +1329,63 @@ void ipc_cmd(struct ipc_cmd_hdr *_hdr)
 		char *data = ipc->comp_data;
 		struct ipc4_message_reply reply;
 
+		/* Process flow and time stamp for IPC4 msg processed on secondary core :
+		 * core 0 (primary core)				core x (secondary core)
+		 * # IPC msg thread		#IPC delayed worker     #core x idc thread
+		 * ipc_task_ops.run()
+		 * ipc_do_cmd()
+		 * msg_reply.header = in->primary.dat
+		 * ipc4_process_on_core(x)
+		 * mask |= SECONDARY_CORE
+		 * idc_send_message()
+		 * Case 1:
+		 * // Ipc msg processed by secondary core		idc_ipc()
+		 * if ((mask & SECONDARY_CORE))				ipc_cmd()
+		 *	return;						ipc_msg_send()
+		 *							mask &= ~SECONDARY_CORE
+		 *
+		 *				ipc_platform_send_msg
+		 * ----------------------------------------------------------------------------
+		 * Case 2:
+		 *                                                      idc_ipc()
+		 *                                                      ipc_cmd()
+		 *							//Prepare reply msg
+		 *                                                      msg_reply.header =
+		 *                                                      reply.primary.dat;
+		 *                                                      ipc_msg_send()
+		 *                                                      mask &= ~SECONDARY_CORE
+		 *
+		 * if ((mask & IPC_TASK_SECONDARY_CORE))
+		 *	return;
+		 * // Ipc reply msg was prepared, so return
+		 * if (msg_reply.header != in->primary.dat)
+		 *	return;
+		 *				ipc_platform_send_msg
+		 * ----------------------------------------------------------------------------
+		 * Case 3:
+		 *                                                      idc_ipc()
+		 *                                                      ipc_cmd()
+		 *							//Prepare reply msg
+		 *                                                      msg_reply.header =
+		 *                                                      reply.primary.dat;
+		 *                                                      ipc_msg_send()
+		 *                                                      mask &= ~SECONDARY_CORE
+		 *
+		 *                              ipc_platform_send_msg
+		 *
+		 * if ((mask & IPC_TASK_SECONDARY_CORE))
```

Comment:
@RanderWang  Case 2 and 3 are identical, I can't spot any difference?

Reply:
@kv2019i I think the difference is when `ipc_platform_send_msg()` is run by the IPC work handler - before core 0 resumes IPC processing or after?